jobs/integration/tigera_aws.py

```diff
@@ -452,22 +454,25 @@ def deploy_bgp_router():
     juju_wait(timeout=15 * 60)
 
     log("Enabling secondary network interfaces")
-    expected_nics = {f"enp{i + 39}s0" for i in range(0, len(subnets))}
+    num_expected_nics = len(subnets)
+    nic_pattern = re.compile("^enp.+s0$")
+    nic_matches = set()
     max_retries, current_nics = 10, set()
-    while not expected_nics.issubset(current_nics) and max_retries:
+    while not len(nic_matches) == num_expected_nics and max_retries:
         time.sleep(10 - max_retries)
         current_nics = {
             line.split("  ", 1)[0]
             for line in juju(
                 "ssh", "router/0", "ip", "-br", "link", "show"
             ).splitlines()
         }
+        nic_matches = {nic for nic in current_nics if nic_pattern.match(nic)}
         max_retries -= 1
 
-    if not expected_nics.issubset(current_nics):
+    if not len(nic_matches) == num_expected_nics:
         log(
             "Failed to find all nics\n"
-            f"  expected: {', '.join(expected_nics)}\n"
+            f"  expected {num_expected_nics} nics matching '{nic_pattern.pattern}'\n"
             f"  current : {', '.join(current_nics)}"
         )
         raise TimeoutError()
```

Comment:
Look 2 lines below this one.  I believe `expected_nics` will now be undefined right?  Maybe it should be `nic_matches`